Cleanup and modernize (#5)
ferrous-systems/teaching-material: presentations/cargo/slides.adoc

Test use the debug profile per default.

> `cargo test --all --examples` runs all test, including documentation test and compilation of examples!

Reviewer: ```suggestion
`cargo test --all --examples` runs all tests, including documentation test and compilation of examples!
```

== `cargo bench`

`cargo bench` runs all benchmarks.

Benchmarks default to the release profile.

> Benchmarks only work on the nightly compiler.

Reviewer: Is it worth mentioning `criterion` here for stable benches?

Author: We probably should.

Author: ```suggestion
Benchmarks can be written using the `criterion` crate.
```

At the first build (or through `cargo update`), cargo calculates a version tree fulfilling the constraints stated in Cargo.toml. If successful, it will be saved in a Lockfile (Cargo.lock).

> It can be considered to always keep the Lockfile under version control! (for repeatable builds)

Reviewer: If you want this statement, I would reword as such:

> For repeatable builds, consider keeping the Lockfile under version control

However, this doesn't totally match what I would call "best practice", which would be:

* For binaries, include lock file (maybe delete directly in some nightly CI testing)
* For libraries, *don't* include a lock file

Author: The problem is that the best practice is debatable. If you don't include the lockfile, you lose all info about what your library was tested with during a test run.

Author: ```suggestion
The Lockfile should always be kept under version control for applications, for libraries, it can be checked in to provide a repeatable test environment.
```

Author: @jamesmunns ^^ Better?
